proposals/XX-bw-xxx.txt

```diff
@@ -0,0 +1,169 @@
+  on relays' measured bandwidth capacities.
+
+2. Format details
+
+  Bandwidth measurements MUST contain the following sections:
+  - Header (exactly once)
+  - Relays measurements (zero or more times)
+
+2.1. Nonterminals
+
+  The following nonterminals are defined in dir-spec.txt, sections
+  1.2., 2.1.1., 2.1.3. :
+
+    Int
+    SP (space)
+    NL (newline)
+    Keyword
+    ArgumentChar
+    fingerprint (hexdigest)
+    nickname
+
+  Nonterminals defined in "Tor Directory List Format" (dir-list-spec.txt),
+  section 2.2.1.:
+
+    version_number
+
+[juga: this needed?
+  We define the following nonterminals:
+
+    value ::= ArgumentChar+
+    key_value ::= Keyword "=" value
+    line := ArgumentChar* NL
+]
+
+2.2. Header format
+
+It consists of:
+
+  timestamp NL
+
+    [At start, exactly once.]
+
+    timestamp = Int
+
+    The Unix Epoch time when the file was created.
+
+  "version="version_number NL
+
+    [In second position, zero or one time.]
+
+    The specification document format version.
+
+    New implementations SHOULD include this line.
+    For documents generated following this specification,
+    the version_number MUST be "1.1.0".
+    Documents generated before this specification do not contain this line,
+    and the version_number is considered to be "1.0.0".
+    This line is ignored by any Tor version released before this specification,
+    April 2018.
+
+  Future releases may include additional header lines.
+  Parsers MUST NOT rely on the order of these additional lines.
+  Additional header lines will be accompanied by a minor version increment.
+  If a header line does conform to this format, the line SHOULD be ignored by
+  parsers.
+
+  [juga: how the header ends?, just when a relay measurement is found?]
+
+2.3. Relay measurements format
+
+  It consists of zero or more lines that contain the measurement results
+  of relays in arbitrary order.
+  There can be at most one such line per relay identity (fingerprint).
+  Each line MUST consists of the following fields in arbitrary order:
+
+    "node_id=" fingerprint SP
+    "bw=" bandwidth SP
+
+  Where:
+
+    bandwidth = Int
+
+    The aggregated measured bandwidth of this relay, in kilobytes
+    per second.
+
+  Each line MUST accept additional Arguments as stated in the dir-spec.txt
+  Document meta format.[juga: should we use the commented definitions instead?]
+  If additional Arguments are included, they MUST have the equal sign ("=")
+  in it.
+  Each line MUST end with NL.
+
+  For instance, every relay measurement in sbws version 0.1.0 consist of:
+
+    "node_id=" fingerprint SP
+    "bw=" bandwidth SP
+    "nick=" nickname SP
+    "rtt=" rtt SP
+    "time=" timestamp NL
+
+  Where:
+
+    timestamp = Int
+
+    The Unix Epoch time when the measurement was performed.
+
+    rtt = Int
+
+    The Round Trip Time to obtain 1 Byte of data.
+
+  [juga: which arguments should be include from torflow?]
```

Comment:
Please include node_id, bw, nick.
I don't think there is a lot of value in documenting legacy systems.

Reply:
Also, there is an example in the appendix, and you have referenced the documentation.